### PR TITLE
7903357: add a jextract sample that uses pcre2 library

### PR DESCRIPTION
**Pull request by sundararajana**
Simple sample that checks a given string for matches against the given regex

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed (no review required)

### Issue
 * [CODETOOLS-7903357](https://bugs.openjdk.org/browse/CODETOOLS-7903357): add a jextract sample that uses pcre2 library

### Reviewers
 * [Jorn Vernee](https://openjdk.org/census#jvernee) (@JornVernee - no project role) ⚠️ Review applies to [d41707c6](https://git.openjdk.org/jextract/pull/88/files/d41707c6364b9cf4317139b245bac028fcb37a85)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jextract pull/88/head:pull/88` \
`$ git checkout pull/88`

Update a local copy of the PR: \
`$ git checkout pull/88` \
`$ git pull https://git.openjdk.org/jextract pull/88/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 88`

View PR using the GUI difftool: \
`$ git pr show -t 88`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jextract/pull/88.diff">https://git.openjdk.org/jextract/pull/88.diff</a>

</details>

### CODE REVIEW

File: samples/pcre2/Pcre.h [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#define PCRE2_CODE_UNIT_WIDTH 8
```
Comment (JornVernee) [inline samples/pcre2/Pcre.h:32]: Couldn't this be done with `-DPCRE2_CODE_UNIT_WIDTH=8` as well?

File: samples/pcre2/PcreCheck.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+          errcodePtr, erroffPtr, NULL);
+
+      // if compilation failed, report error and exit
+      if (re.equals(NULL)) {
+        var buffer = session.allocateArray(C_CHAR, 128L);
+        pcre2_get_error_message_8(errcodePtr.get(C_INT, 0),
+            buffer, 127);
+        System.err.printf("regex compilation failed: %s\n",
+            buffer.getUtf8String(0));
+        pcre2_code_free_8(re);
```
Comment (JornVernee) [inline samples/pcre2/PcreCheck.java:63]: Does this free do anything? `re` seems to be `NULL` here.

**Review by JornVernee — APPROVED** [review]
Looks mostly good

File: samples/pcre2/PcreCheck.java (hunk 2 above)
Comment (sundararajana) [reply samples/pcre2/PcreCheck.java]: will fix

**Pull request closed without merging**